content/tutorials/r_gbif_checklist/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+```r
+distribution <- name_usage(datasetKey = datasetKey, data = "distribution")
+Error in exists(z, private$crul_h_pool) : invalid first argument
+```
+
+You need a loop over all the taxa (`key`):
+
+```{r get_distribution}
+distribution <- 
+  red_list_taxa %>% 
+  pull(key) %>%
+  map_dfr(., function(x) {
+    distribution_details <-name_usage(x, data = "distribution")
+    return(distribution_details$data)
+    }
+  )
+```
+
+Overview:
+
+```{r}
+kable(distribution[1:10,])
+```
+
+This step is slower than retrieving the taxonomic information as we need 3036 calls to the API, one per taxon, instead of one.
+
+Again, can you find this information on the GBIF website? Of course! All infos about the extensions is shown on the taxon page. Below a screenshot showing the section related to the distribution extension for [_Plea minutissima_](https://www.gbif.org/species/152624576)(`key`: `152624576`):
+
+![](gbif_webpage_extension.png)
+
+## Cite the checklist dataset
+
+If the (checklist) dataset you used is worth to be cited, you can retrieve the citation by running `gbif_citation()` function:
+
+```{r cite}
+gbif_citation(x = datasetKey)
+```
+
+## INBO datasets on GBIF
+
+INBO is an official GBIF publisher and a host for many other smaller publishers. Do you want to know which datasets have been published so far by INBO? 
+You can have a look via this link:
+
+https://www.gbif.org/dataset/search?publishing_org=1cd669d0-80ea-11de-a9d0-f1765f95f18b
+
+you can finetuning the search by selecting the type of the datasets. For example, for checklist datasets:
+
+https://www.gbif.org/dataset/search?type=CHECKLIST&publishing_org=1cd669d0-80ea-11de-a9d0-f1765f95f18b
+
+Can you do this in R? Of course. Use the rgbif function `dataset_search()` specifying the INBO's ID as publishing organization. Again, you get a list with two slots, `meta` and `data`. Guess where the datasets are :-)
+
+```{r get_datasets_INBO}
+INBO_datasets <- dataset_search(
+  publishingOrg = "1cd669d0-80ea-11de-a9d0-f1765f95f18b"
+)
+INBO_datasets$data
+```
+
+To finetune the search as we did online, use `type` argument:
+
+```{r inbo_checklists}
+INBO_checklists <- dataset_search(
+  publishingOrg = "1cd669d0-80ea-11de-a9d0-f1765f95f18b",
+  type = "checklist"
+)
+INBO_checklists$data 
+```
+
+## INBO as hosting organization
+
+INBO is not only a GBIF publisher. It hosts GBIF data for many other organisations. How to get all hosted datasets? By selecting INBO in the field "Host":
+
+Online: https://www.gbif.org/dataset/search?hosting_org=1cd669d0-80ea-11de-a9d0-f1765f95f18b
+
+In R, by using the argument `hostingOrg` within function `dataset_search()`:
+
+```{r}
+datasets_hosted_by_inbo <- dataset_search(
+  hostingOrg = "1cd669d0-80ea-11de-a9d0-f1765f95f18b",
+  limit = 1000)
+datasets_hosted_by_inbo$data
+```
+
+Notice again the `limit` argument with a sufficiently high number as value. Exercise: check the `meta` slot to check if we got all records.
```

Comment:
Does a method exist to a-priori find the number of 'hits' so the limit can be confidently set high enough?

Reply:
Actually no. The method is setting a very high number to be safe and check the `meta` slot to check you got them all.
Or you do the search first on GBIF web portal so you know how many datasets to expect.